app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js

```diff
@@ -185,6 +185,8 @@ export async function switchChain(
       }
     }
 
+    // await new Promise((resolve) => setTimeout(resolve, 1000))
```

Comment:
TODO: remove this. Leaving it here to highlight what the hacky alternative is to dealing with making sure requests queued behind an add/switch chain make it into pending so that they are cleared by the setActiveNetwork that happens below this